### PR TITLE
sqlite: Don't use multiple handles to the same file

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Part of: https://github.com/bpfkorea/agora/issues/836

### CODE REVIEW

File: source/agora/consensus/EnrollmentManager.d [hunk 1]
```diff
@@ -146,10 +146,10 @@ public class EnrollmentManager
             /* preimages: */ PreImageCache(this.params.ValidatorCycle, 1)
         );
 
-        this.validator_set = new ValidatorSet(db_path, params);
-        this.enroll_pool = new EnrollmentPool(db_path);
-
         this.db = new ManagedDatabase(db_path);
+        this.validator_set = new ValidatorSet(this.db, params);
```
Comment (AndrejMitrovic) [inline source/agora/consensus/EnrollmentManager.d:150]: Alternatively, maybe these should be separate files. But I'm not sure what's more ideal here.

Comment (TrustHenry) [reply source/agora/consensus/EnrollmentManager.d:150]: How about applying it in WAL(Write-Ahead Logging) option?
https://www.sqlite.org/wal.html

Comment (AndrejMitrovic) [reply source/agora/consensus/EnrollmentManager.d:150]: It seems somewhat unrelated. It enables a reader and writer at the same time, but in our case we have multiple writers to the same file.

Anyway it's worth looking into WAL later, sure.

**Comment by Geod24** [comment]
Tested, works!

**Review by Geod24 — APPROVED** [review]

**Pull request merged**